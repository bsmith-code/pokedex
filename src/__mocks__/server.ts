import {
  mockPokemonDetails,
  mockPokemonEvolutions,
  mockPokemonList,
  mockPokemonSpecies
} from '__mocks__/pokemon'
import { http, HttpResponse } from 'msw'
import { setupServer } from 'msw/node'

export const prepareRoute = (route: string) =>
  `${process.env.REACT_APP_API_BASE_URL ?? ''}/${route}`

const handlers = [
  http.get(prepareRoute('pokemon'), () => HttpResponse.json(mockPokemonList)),
  http.get(prepareRoute('pokemon/:name'), () =>
    HttpResponse.json(mockPokemonDetails)
  ),
  http.get(prepareRoute('pokemon-species/:name'), () =>
    HttpResponse.json(mockPokemonSpecies)
  ),
  http.get(prepareRoute('evolution-chain/:chainId'), () =>
    HttpResponse.json(mockPokemonEvolutions)
  )
]

export const mockServer = setupServer(...handlers)
