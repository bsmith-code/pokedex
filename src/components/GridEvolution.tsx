import { IEvolutionChain } from 'pokeapi-typescript'

import { Box, Grid, ListItemButton, styled, Typography } from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { flattenChain, getIdFromUrl, getPokemonSprite } from 'utils'

interface IProps {
  isFetching: boolean
  evolutions?: IEvolutionChain
}

const StyledGridItem = styled(ListItemButton)(({ theme }) => ({
  flexDirection: 'column',
  borderRadius: '4px',
  color: '#fff',
  padding: theme.spacing(4),
  backgroundColor: theme.palette.secondary.main,
  '&:hover': {
    backgroundColor: '#6704c9'
  }
}))

export const GridEvolution = ({ evolutions, isFetching }: IProps) => {
  const { handleChangePokemon } = usePokemonContext()
  const flattenedChain = flattenChain(evolutions?.chain)

  return !isFetching ? (
    <Box>
      <Typography variant="h6" sx={{ mb: 2 }}>
        Evolutions
      </Typography>
      {flattenedChain?.length ? (
        <Grid container spacing={2}>
          {flattenedChain.map(({ url, name }) => {
            const id = getIdFromUrl(url)
            return (
              <Grid item xs={12} sm={6} md={6} lg={4} key={`evolution-${name}`}>
                <StyledGridItem onClick={() => handleChangePokemon(name)}>
                  <img
                    alt={name}
                    src={getPokemonSprite(id)}
                    width="120px"
                    height="120px"
                    style={{ objectFit: 'contain' }}
                  />
                  <Typography variant="subtitle2">{name}</Typography>
                </StyledGridItem>
              </Grid>
            )
          })}
        </Grid>
      ) : (
        <Typography>No available evolutions</Typography>
      )}
    </Box>
  ) : (
    <Typography>Loading...</Typography>
  )
}
