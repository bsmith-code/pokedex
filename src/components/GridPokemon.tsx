import {
  Box,
  Grid,
  ListItemButton,
  Pagination,
  styled,
  Typography
} from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { getIdFromUrl, getPokemonSprite } from 'utils'

import {
  TESTID_GRID_ERROR,
  TESTID_GRID_LOADING,
  TESTID_GRID_POKEMON
} from 'constants/testIds'

const StyledGridItem = styled(ListItemButton)`
  flex-direction: column;
  border-radius: 4px;
  border: 1px solid #efefef;
`

export const GridPokemon = () => {
  const {
    pokemon,
    currentPage,
    paginatedPokemon,
    isErrorPokemon,
    isFetchingPokemon,
    handleChangePage,
    handleChangePokemon
  } = usePokemonContext()

  if (isFetchingPokemon) {
    return <div data-testid={TESTID_GRID_LOADING}>Loading...</div>
  }

  if (!pokemon.length || isErrorPokemon) {
    return <div data-testid={TESTID_GRID_ERROR}>Pokemon unavailable.</div>
  }

  return (
    <Box mt={4} data-testid={TESTID_GRID_POKEMON}>
      <Grid container spacing={2}>
        {paginatedPokemon.map(({ url, name }) => {
          const imageSrc = getPokemonSprite(getIdFromUrl(url))

          return (
            <Grid
              item
              xs={12}
              sm={6}
              md={4}
              lg={3}
              key={`pokemon-grid-${name}`}
            >
              <StyledGridItem onClick={() => handleChangePokemon(name)}>
                <img
                  alt={name}
                  src={imageSrc}
                  width="120px"
                  height="120px"
                  style={{ objectFit: 'contain' }}
                />
                <Typography>{name}</Typography>
              </StyledGridItem>
            </Grid>
          )
        })}
      </Grid>
      <Box mt={6} display="flex" justifyContent="center">
        <Pagination
          page={currentPage}
          count={Math.floor(pokemon.length / 20)}
          onChange={(_, page) => handleChangePage(page)}
        />
      </Box>
    </Box>
  )
}
