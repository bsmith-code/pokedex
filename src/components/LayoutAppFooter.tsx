import { Toolbar, Typography } from '@mui/material'

const LayoutAppFooter = () => {
  const year = new Date().getFullYear()

  return (
    <Toolbar
      variant="dense"
      component="footer"
      sx={{ bgcolor: 'primary.main', justifyContent: 'center' }}
    >
      <Typography color="#fff">
        &copy; Brian Matthew Smith, {year}. All Rights Reserved.
      </Typography>
    </Toolbar>
  )
}

export default LayoutAppFooter
