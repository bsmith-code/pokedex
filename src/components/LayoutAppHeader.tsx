import { useNavigate } from 'react-router-dom'

import { IconButton, Toolbar } from '@mui/material'

const LayoutAppHeader = () => {
  const navigate = useNavigate()

  const handleLogoClick = () => {
    navigate('/')
  }

  return (
    <Toolbar component="header" sx={{ bgcolor: 'primary.main' }}>
      <IconButton onClick={handleLogoClick}>
        <img
          alt="PrizePicks"
          src="/assets/logo.svg"
          width="140px"
          height="auto"
        />
      </IconButton>
    </Toolbar>
  )
}

export default LayoutAppHeader
