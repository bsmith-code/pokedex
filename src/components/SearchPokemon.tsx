import { INamedApiResource } from 'pokeapi-typescript'

import {
  Autocomplete,
  createFilterOptions,
  ListItemButton,
  TextField
} from '@mui/material'

import { useChangePokemon } from 'hooks/useChangePokemon'

interface IProps {
  pokemon: INamedApiResource<void>[]
}

export const SearchPokemon = ({ pokemon }: IProps) => {
  const { handleChangePokemon } = useChangePokemon()

  return (
    <Autocomplete
      filterOptions={createFilterOptions({ limit: 20 })}
      onChange={(_, value) => handleChangePokemon(value?.name ?? '')}
      options={pokemon}
      renderInput={params => <TextField {...params} label="Search Pokemon" />}
      getOptionLabel={({ name }) => name}
      renderOption={(props, { name }) => (
        <ListItemButton
          {...props}
          component="li"
          key={`pokemon-list-${name}`}
          data-testid={`pokemon-${name}`}
        >
          {name}
        </ListItemButton>
      )}
    />
  )
}
