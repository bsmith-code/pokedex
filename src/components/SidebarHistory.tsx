import { Box, List, ListItemButton, Paper, Typography } from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { TESTID_SEARCH_EMPTY, TESTID_SEARCH_HISTORY } from 'constants/testIds'

export const SidebarHistory = () => {
  const { searchHistory, handleChangePokemon } = usePokemonContext()

  return (
    <Box
      px={4}
      py={3}
      flexShrink={1}
      flexBasis={300}
      component={Paper}
      bgcolor="grey.200"
      sx={{ '@media (max-width: 767px)': { order: 2, flexGrow: 1 } }}
    >
      <Typography variant="subtitle2">Search History</Typography>

      {searchHistory.length ? (
        <List data-testid={TESTID_SEARCH_HISTORY}>
          {searchHistory.map(name => (
            <ListItemButton
              component="li"
              key={`search-${name}`}
              onClick={() => handleChangePokemon(name)}
            >
              {name}
            </ListItemButton>
          ))}
        </List>
      ) : (
        <Typography data-testid={TESTID_SEARCH_EMPTY} sx={{ mt: 2 }}>
          No previous searches.
        </Typography>
      )}
    </Box>
  )
}
