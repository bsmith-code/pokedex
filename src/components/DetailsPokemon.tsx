import { useNavigate } from 'react-router-dom'

import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import {
  Box,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  Typography
} from '@mui/material'

import { usePokemonDetails } from 'hooks/usePokemonDetails'

import { GridEvolution } from 'components/GridEvolution'

import {
  TESTID_DETAILS_ERROR,
  TESTID_DETAILS_LOADING,
  TESTID_DETAILS_POKEMON,
  TESTID_DETAILS_RETURN
} from 'constants/testIds'

export const DetailsPokemon = () => {
  const {
    name,
    details,
    sprites,
    evolutions,
    speciesName,
    isErrorDetails,
    isFetchingDetails,
    isFetchingEvolutions
  } = usePokemonDetails()
  const navigate = useNavigate()

  const handleGoBack = () => {
    navigate('/')
  }

  if (isFetchingDetails) {
    return <div data-testid={TESTID_DETAILS_LOADING}>Loading...</div>
  }

  if (isErrorDetails) {
    return <div data-testid={TESTID_DETAILS_ERROR}>Invalid Pokemon.</div>
  }

  return (
    <>
      <Box display="flex" mt={2} data-testid={TESTID_DETAILS_POKEMON}>
        <Box flexShrink={0} position="relative">
          <IconButton
            onClick={handleGoBack}
            data-testid={TESTID_DETAILS_RETURN}
            sx={{ position: 'absolute', top: 0, left: 0 }}
          >
            <ArrowBackIcon />
          </IconButton>
          <img
            alt={name}
            src={sprites?.other?.['official-artwork']?.front_default ?? ''}
          />
        </Box>
        <Box p={2} flexGrow={1}>
          <Box mb={2}>
            <Typography component="h1" variant="h4">
              {name}
            </Typography>
            <Typography fontSize={14} color="grey.700" sx={{ mt: 1 }}>
              species: {speciesName}
            </Typography>
          </Box>
          {details.map(({ title, content }) => (
            <Card
              key={`details-${title}`}
              sx={{ bgcolor: 'secondary.main', color: 'common.white', mb: 2 }}
            >
              <CardHeader title={title} />
              <CardContent>{content}</CardContent>
            </Card>
          ))}
        </Box>
      </Box>
      <GridEvolution
        evolutions={evolutions}
        isFetching={isFetchingEvolutions}
      />
    </>
  )
}
