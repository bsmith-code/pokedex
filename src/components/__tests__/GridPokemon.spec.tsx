import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { mockServer, prepareRoute } from '__mocks__/server'
import { fireEvent, screen, waitFor, within } from '@testing-library/react'
import { http, HttpResponse } from 'msw'

import { PokemonContextProvider } from 'context/PokemonContext'

import { GridPokemon } from 'components/GridPokemon'

import {
  TESTID_GRID_ERROR,
  TESTID_GRID_LOADING,
  TESTID_GRID_POKEMON
} from 'constants/testIds'

const MockComponent = () => (
  <BrowserRouter>
    <Routes>
      <Route
        index
        path="/pokemon?/:pokemonName?"
        element={
          <PokemonContextProvider>
            <GridPokemon />
          </PokemonContextProvider>
        }
      />
    </Routes>
  </BrowserRouter>
)

describe('GridPokemon', () => {
  it('renders correctly while loading', async () => {
    const { container } = customRender(<MockComponent />)

    await waitFor(() => {
      expect(screen.getByTestId(TESTID_GRID_LOADING)).toBeInTheDocument()
      expect(container).toMatchSnapshot()
    })
  })

  it('renders correctly with data', async () => {
    const { container } = customRender(<MockComponent />)

    await waitFor(() => {
      expect(screen.getByTestId(TESTID_GRID_POKEMON)).toBeInTheDocument()
      expect(container).toMatchSnapshot()
    })
  })

  it('renders correctly with invalid data', async () => {
    mockServer.use(
      http.get(prepareRoute('pokemon'), () => HttpResponse.error())
    )
    const { container } = customRender(<MockComponent />)

    await waitFor(() => {
      expect(screen.getByTestId(TESTID_GRID_ERROR)).toBeInTheDocument()
      expect(container).toMatchSnapshot()
    })
  })

  it('correctly changes pokemon', async () => {
    customRender(<MockComponent />)

    const pokemon = await screen.findByTestId(TESTID_GRID_POKEMON)
    const pokemonBtn = within(pokemon).getAllByRole('button')[0]
    fireEvent.click(pokemonBtn)

    await waitFor(() => {
      expect(window.location.pathname).toBe(
        `/pokemon/${pokemonBtn.textContent ?? ''}`
      )
    })
  })
})
