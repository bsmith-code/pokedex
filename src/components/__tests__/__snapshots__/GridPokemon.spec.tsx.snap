// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridPokemon renders correctly while loading 1`] = `
<div>
  <div
    data-testid="grid-loading"
  >
    Loading...
  </div>
</div>
`;

exports[`GridPokemon renders correctly with data 1`] = `
<div>
  <div
    class="MuiBox-root emotion-0"
    data-testid="grid-pokemon"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 emotion-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-0"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/0.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-0
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-1"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-1
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-2"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-2
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-3"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-3
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-4"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-4
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-5"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/5.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-5
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-6"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-6
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-7"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/7.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-7
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-8"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/8.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-8
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-9"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/9.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-9
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-10"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-10
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-11"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/11.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-11
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-12"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/12.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-12
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-13"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/13.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-13
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-14"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/14.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-14
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-15"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/15.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-15
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-16"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/16.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-16
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-17"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/17.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-17
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-18"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/18.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-18
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-2"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-3"
          role="button"
          tabindex="0"
        >
          <img
            alt="pokemon-19"
            height="120px"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/19.png"
            style="object-fit: contain;"
            width="120px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            pokemon-19
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root emotion-62"
    >
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text emotion-63"
      >
        <ul
          class="MuiPagination-ul emotion-64"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext emotion-65"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon emotion-66"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page emotion-65"
              tabindex="0"
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-65"
              tabindex="0"
              type="button"
            >
              2
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-65"
              tabindex="0"
              type="button"
            >
              3
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-65"
              tabindex="0"
              type="button"
            >
              4
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 5"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-65"
              tabindex="0"
              type="button"
            >
              5
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext emotion-65"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon emotion-66"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`GridPokemon renders correctly with invalid data 1`] = `
<div>
  <div
    data-testid="grid-error"
  >
    Pokemon unavailable.
  </div>
</div>
`;
